metrics/register.go

```diff
@@ -0,0 +1,38 @@
+package metrics
+
+import (
+	"fmt"
+
+	"go.opencensus.io/stats/view"
+)
+
+var registeredViews = map[string][]*view.View{}
```

Comment:
This map should be made threadsafe via a mutex.

Reply:
Also document this type; it's not obvious that the string is actually a namespace.

Reply:
Updated the documentation to better describe what the map is storing. Also changed it to `sync.Map{}`, is that acceptable?

Reply:
A Mutex is the standard go bread and butter for sync. Did you need something specific from sync.Map?

Reply:
Not sure. It seemed more appropriate, because of this line ([from godocs](https://golang.org/pkg/sync/#Map)): 
> The Map type is optimized for two common use cases: (1) when the entry for a given key is only ever written once but read many times, as in caches that only grow

Seeing as how this is a write only map, it seemed to make sense. If you prefer the standard map for type safety, I can write it back.

Reply:
@raulk thoughts?

Reply:
@raulk wrote it back to use sync.Mutex, I don't want this to get stuck